hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/JavaDefaultSerializers.java

```diff
@@ -67,12 +69,17 @@ public Externalizable read(final ObjectDataInput in) throws IOException {
                 final Externalizable ds = ClassLoaderUtil.newInstance(in.getClassLoader(), className);
                 final ObjectInputStream objectInputStream;
                 final InputStream inputStream = (InputStream) in;
+                ExtendedGZipInputStream gZipInputStream = null;
                 if (gzipEnabled) {
-                    objectInputStream = newObjectInputStream(in.getClassLoader(), new GZIPInputStream(inputStream));
+                    gZipInputStream = new ExtendedGZipInputStream(inputStream);
```

Comment:
Seems not a side effect of this PR but shouldn't we close `gZipInputStream`? Remembered this one: http://www.evanjones.ca/java-native-leak-bug.html

Reply:
that would also close the inner stream and that's not what we want. but this line should prevent the leak: https://github.com/hazelcast/hazelcast/pull/12099/files#diff-47cd284d0b603ce35aad0b8ffea162d5R327